### PR TITLE
[bazel] upgrade cython package

### PR DESCRIPTION
**Pull request by aslonnie**
to 0.29.37; required for bazel upgrade.

### CODE REVIEW

**Comment by aslonnie** [comment]
this builds and passes on premerge, but when compiling locally, it says errors like this:

````
File 'ExprNodes.py', line 7478, in _create_unbound_cmethod_entry: AttributeNode(_raylet.pyx:4563:64,
    attribute = 'Nil',
    initialized_check = True,
    is_attribute = 1,
    is_called = 1,
    needs_none_check = True,
    result_is_used = True,
    use_managed_ref = True)

Compiler crash traceback from this point on:
  File "/home/ubuntu/.cache/bazel/_bazel_ubuntu/211c11eff41c965c8c193a74ddf9e96b/external/cython/Cython/Compiler/ExprNodes.py", line 7478, in _create_unbound_cmethod_entry
    cname = "%s->%s" % (type.vtabptr_cname, entry.cname)
AttributeError: 'CppClassType' object has no attribute 'vtabptr_cname'
````

**Comment by aslonnie** [comment]
````
Error compiling Cython file:
------------------------------------------------------------
...
    cdef store_task_outputs(self,
                            worker, outputs,                                                                                                                                            const CAddress &caller_address,
                            c_vector[c_pair[CObjectID, shared_ptr[CRayObject]]]
                            *returns,
                            CObjectID ref_generator_id=CObjectID.Nil()):
                                                                ^
------------------------------------------------------------

python/ray/_raylet.pyx:4563:64: Compiler crash in AnalyseExpressionsTransform

File 'Nodes.py', line 397, in analyse_expressions: StatListNode(_raylet.pyx:7:0)
File 'Nodes.py', line 397, in analyse_expressions: StatListNode(_raylet.pyx:3283:5)
File 'Nodes.py', line 5547, in analyse_expressions: CClassDefNode(_raylet.pyx:3283:5,
    as_name = 'CoreWorker',
    class_name = 'CoreWorker',
    module_name = '',
    punycode_class_name = 'CoreWorker',
    visibility = 'private')
File 'Nodes.py', line 397, in analyse_expressions: StatListNode(_raylet.pyx:3285:4)
File 'Nodes.py', line 2844, in analyse_expressions: CFuncDefNode(_raylet.pyx:4558:9,
    args = [...]/6,
    is_c_class_method = 1,
    modifiers = [...]/0,
    outer_attrs = [...]/2,
    visibility = 'private')
File 'Nodes.py', line 1896, in analyse_default_values: CFuncDefNode(_raylet.pyx:4558:9,
    args = [...]/6,
    is_c_class_method = 1,
    modifiers = [...]/0,
    outer_attrs = [...]/2,
    visibility = 'private')
File 'ExprNodes.py', line 6099, in analyse_types: SimpleCallNode(_raylet.pyx:4563:68,
    analysed = True,
    result_is_used = True,
    use_managed_ref = True)
File 'ExprNodes.py', line 7398, in analyse_types: AttributeNode(_raylet.pyx:4563:64,
    attribute = 'Nil',
    initialized_check = True,
    is_attribute = 1,
    is_called = 1,
    needs_none_check = True,
    result_is_used = True,
    use_managed_ref = True)
File 'ExprNodes.py', line 7445, in analyse_as_type_attribute: AttributeNode(_raylet.pyx:4563:64,
    attribute = 'Nil',
    initialized_check = True,
    is_attribute = 1,
    is_called = 1,
    needs_none_check = True,
    result_is_used = True,
    use_managed_ref = True)
File 'ExprNodes.py', line 7446, in <listcomp>: AttributeNode(_raylet.pyx:4563:64,
    attribute = 'Nil',
    initialized_check = True,
    is_attribute = 1,
    is_called = 1,
    needs_none_check = True,
    result_is_used = True,
    use_managed_ref = True)
````

**Comment by aslonnie** [comment]
seems that I need to stick with 0.29.x

File: rllib/tests/git_bisect/debug_learning_failure_git_bisect.py [hunk 1]
```diff
@@ -183,7 +183,7 @@
         subprocess.run(
             "sudo apt-get install -y build-essential curl unzip psmisc".split(" ")
         )
-        subprocess.run("pip install cython==0.29.26 pytest".split(" "))
+        subprocess.run("pip install cython==0.29.37 pytest".split(" "))
```
Comment (aslonnie) [inline rllib/tests/git_bisect/debug_learning_failure_git_bisect.py:186]: @sven1977 , do the rllib people still use this bisect tool?

fwiw, we have repo wide bisect tool now.

**Review by can-anyscale — APPROVED** [review]
tests passed ;)

**Comment by aslonnie** [comment]
ping for review.

**Comment by aslonnie** [comment]
hoping to get review from at least @jjyao and @sven1977

**Review by jjyao — APPROVED** [review]
Thanks a lot for this!!

**Comment by aslonnie** [comment]
ok I am merging. @sven1977 (or any one else), if you need to revert this or some files in this, let me know.

**Pull request merged**